src/index.js

```diff
@@ -30,39 +30,44 @@ class ESLintWebpackPlugin {
       files: parseFiles(this.options.files, this.getContext(compiler)),
       extensions: arrify(this.options.extensions),
     };
+    Object.freeze(options);
 
-    if (options.lintDirtyModulesOnly) {
-      const dirtyFileWatcher = new DirtyFileWatcher(
-        options.files,
-        options.extensions
-      );
+    /**
+     * @param {Compiler} runCompiler
+     */
+    const run = async (runCompiler) => {
+      /** @type {string[]} */
+      const files = [];
+      const wanted = parseFoldersToGlobs(options.files, options.extensions);
 
-      /* istanbul ignore next */
-      compiler.hooks.watchRun.tapPromise(
+      runCompiler.hooks.compilation.tap(
```

Comment:
`thisCompilation`, otherwise we will run plugin on child compilations, but we should do not do it

Reply:
what should I do for webpack 4?